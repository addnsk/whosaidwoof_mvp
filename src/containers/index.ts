export * from "./Header";
export * from "./Theme";
export * from "./SelectList";
