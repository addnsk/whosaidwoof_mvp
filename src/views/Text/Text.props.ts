type TextAs = "h1" | "h2" | "h3" | "h4" | "span" | "p";
type TextAppearance = "h1" | "h2" | "h3" | "h4" | "span" | "p" | "buttonText";
type TextColor = "light" | "dark" | "accentLight" | "accentDark" | "neutral";

export interface ITextProps {
  as?: TextAs;
  appearance: TextAppearance;
  caption: string;
  colorText?: TextColor;
}
