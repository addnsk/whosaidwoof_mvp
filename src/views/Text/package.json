{
  "name": "text-view",
  "version": "0.1.0",
  "private": true,
  "main": "Text.tsx"
}
