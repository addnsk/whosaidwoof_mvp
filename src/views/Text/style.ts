import styled from "styled-components";
import { ITextProps } from "./Text.props";

export const Typography = styled.span<ITextProps>`
  display: inline;
  margin: 0;
  padding: 0;
  color: ${({ colorText, theme }) =>
    colorText ? theme.colors[colorText] : "inherit"};
  ${({ appearance, theme }) => appearance && theme.typography[appearance]}
`;
