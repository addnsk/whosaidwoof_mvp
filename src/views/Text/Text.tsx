import * as React from "react";
import { Typography } from "./style";
import { ITextProps } from "./Text.props";

export const Text: React.FC<ITextProps> = React.memo(
  ({ as = "span", appearance, colorText, caption, ...props }): JSX.Element => {
    return (
      <Typography
        as={as}
        appearance={appearance}
        colorText={colorText}
        caption={caption}
        {...props}
      >
        {caption}
      </Typography>
    );
  }
);
