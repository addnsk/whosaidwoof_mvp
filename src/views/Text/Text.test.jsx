import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Theme } from "containers";
import { Text } from "./Text";

describe("Text component", () => {
  it("Text renders with full props", () => {
    render(
      <Theme>
        <Text as="h2" caption="Hello" appearance="h3" colorText="neutral" />
      </Theme>
    );
    expect(screen.getByText(/hello/i)).toBeInTheDocument();
    expect(screen.getByText(/hello/i).tagName).toEqual("H2");
    expect(screen.getByText(/hello/i)).toHaveStyle({
      color: "#7D808B",
      fontSize: "20px",
    });

    expect(screen.getByText(/hello/i)).toMatchSnapshot();
  });
  it("Text renders with some default props and children", () => {
    render(
      <Theme>
        <Text caption="Hello">Goodbye</Text>
      </Theme>
    );
    expect(screen.getByText(/hello/i)).toBeInTheDocument();
    expect(screen.getByText(/hello/i).tagName).toEqual("SPAN");
    expect(screen.getByText(/hello/i)).toHaveStyle({
      color: "inherit",
      fontSize: "inherit",
    });
    expect(screen.queryByText(/goodbye/i)).toBeNull();
    expect(screen.getByText(/hello/i)).toMatchSnapshot();
  });

  it("Text renders without props", () => {
    render(<Text data-testid="text" />);
    expect(screen.queryByTestId("text")).toBeInTheDocument();
    expect(screen.queryByTestId("text").textContent).toEqual("");
    expect(screen.queryByTestId("text")).toMatchSnapshot();
  });
});
