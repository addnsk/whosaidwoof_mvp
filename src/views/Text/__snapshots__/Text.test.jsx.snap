// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Text renders with full props 1`] = `
<h2
  class="sc-gsTEea lenZgS"
>
  Hello
</h2>
`;

exports[`Text component Text renders with some default props and children 1`] = `
<span
  class="sc-gsTEea eYisTA"
>
  Hello
</span>
`;

exports[`Text component Text renders without props 1`] = `
<span
  class="sc-gsTEea eYisTA"
  data-testid="text"
/>
`;
